text/0000-network-transform-buffered-interpolation.md

```diff
@@ -0,0 +1,447 @@
+
+It also didn't take into account network jitter, which becomes a visible issue on overloaded or unstable networks like you can find with mobile platforms 
+
+The following video shows the MLAPI-0.1.0 NetworkTransform under 5% packet loss and 60-100ms jitter.
+
+https://user-images.githubusercontent.com/71790295/126506051-ebfb1974-e4fc-44ca-9923-2be450c722b6.mp4
+
+![](0000-network-transform-buffered-interpolation/jitter.jpg)
+
+This RFC suggests an extensible interface for users to use and create different interpolation algorithms. It also suggests a default buffered interpolator that'll resist to jitter. 
+After this, users should be able to select the interpolation algorithm that best fits the GameObject they want to sync.
+
+In addition, this RFC suggests removing the 
+```C#
+        public AnimationCurve DistanceSendrate = AnimationCurve.Constant(0, 500, 20);
+```
+configuration (that was used to set interpolation times) to use instead a swappable interpolator.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+NetworkTransform now has this new configuration:
+- `Interpolator`: Allows users to select the interpolation algorithm they want to use. This interpolator will interpolate "ghost" side state changes. Note it'll still be up to users to interpolate their authoritative objects however they want. If for example Client 1 is authoritative over a player object, this RFC will add interpolation to the server object and the other clients' objects, not Client 1's object. It'll still be up to users to interpolate that authoritative object (using RigidBody's interpolation for example or anything else).
+
+
+A default `BufferedLinearInterpolator` is provided. The buffered linear interpolator will buffer values before making them available to NetworkTransform's value update. This will allow NetworkTransform to accumulate jittered network values without affecting the transform's state and latter be able to consume them at regular intervals.
+
+This will smooth tranforms on jittery connections, however it'll add more latency to your transform updates. This should be used carefully as to not add more latency than needed to your game. Bigger buffer values could be expected when targeting mobile platforms for example, while keeping interpolation times lower for platforms with typically more stable connections.
+
+You can find the interpolator's configuration in its associated scriptable object factory.
+It is advised to use the same interpolator for all your NetworkTransforms. This way, you're making sure all your objects will be synchronized and use the same delays. Having an interpolator with 100ms delay and another with 500ms delay could cause visible overlaps and desyncs where an object at time say `t=10s` tries to interact with an object buffered at time `t=9.6s`.
```

Comment:
I believe using the same interpolator and its settings is fair for "same type of NetworkTransforms" but I wouldn't advise to use for "all NetworkTransforms". A car would be interpolated differently compared to a character. It'd make sense to use the same interpolator+config for all cars but it wouldn't make sense to share that interpolator+config with a character walking around.

Reply:
Good point, I'll clarify that. The way I wrote made it seem more like an exception, but like you say these could be very common "exceptions".